File: jieun/Programmers/단속카메라.py
def solution(routes):
    camera = 0          # 카메라 개수
    camera_pos = 30001  # 이전 카메라의 위치
    routes.sort(key=lambda x:x[1])

    for route in routes:
        enter_point, exit_point = route
        if enter_point <= camera_pos <= exit_point:
            continue
        camera_pos = exit_point
        camera += 1

    return camera

File: yoonju/Programmers/Level 3/단속카메라.js
function solution(routes) {
  let answer = 0;
  let lastCamera = -2e9;

  routes.sort((a, b) => a[1] - b[1]);

  routes.forEach(([start, end]) => {
    if (start > lastCamera) {
      answer += 1;
      lastCamera = end;
    }
  });

  return answer;
}
